specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-01-01-preview/workspaceRP.json

```diff
@@ -1906,6 +3074,10 @@
           "type": "object",
           "additionalProperties": {}
         },
+        "hub": {
```

Comment:
>hub [](http://example.com/codeflow?start=9&length=3)

Empty/unschematized object pattern is not allowed. Please provide the schema for this object.

Reply:
Just like the above and below property, this is just a dictionary of object we return the diagnose result to the user, we don't expect user to interact with the content apart from read the diagnose result. can I keep this newly added property same as the existing one's? it will make the user experience consistent.

Reply:
Where does the contents of this property come from? Are the dictionary entries documented anywhere? Are they versioned?

Reply:
the content is from the result of our self diagnose API, which in this case is intend to help the customer to understand if there is any issue with the setup of this resource, the shouldn't expect to interact with any of the results, this is more like helping the CSS engineer and customer to resolve setup issue.

Reply:
https://learn.microsoft.com/en-us/azure/machine-learning/how-to-workspace-diagnostic-api?view=azureml-api-2

in this doc we say it will be a JSON result, and that's what we return in our SDK.

Reply:
ARM requires exact schema for all these resources. Since you own the data you should be able to define an exact schema for this right ?

Reply:
Ok let me remove this for now and I will get back to check the service side code as they are not defined in a clear schema